src/features/device-types/hooks.ts

```diff
@@ -49,6 +50,7 @@ hooks.addPureHook('POST', 'resin', 'application', {
 	},
 });
 
+// TODO: Drop in v7
```

Comment:
What about v7 would allow us to drop this?

Reply:
Just that the major bump allows to do breaking changes.
This looked like a leftover in oB when we replaced the string `device_type` props with the `is_of/for__device_type` device_type resource FKs, which we left on purpose, so that oB doesn't break with a half-changed unreleased model.

More specifically, given that the cloud API also has translations for these exact cases:
See: https://github.com/balena-io/balena-api/blob/e291c4aad2219f7cc5b9e7bfc10e881ffe449a7c/src/models/hooks/translations/v5.ts#L703
See: https://github.com/balena-io/balena-api/blob/e291c4aad2219f7cc5b9e7bfc10e881ffe449a7c/src/models/hooks/translations/v5.ts#L768
and that this is a divergence (arguable helpful though) from how we have so far handled such changes (moving from a primitive field to an FK), it felt to me that we just forgot to drop this once we bumped oB to v6.

I might be wrong though & I don't really remember whether this was discussed and done on purpose.
cc: @dfunckt in case he remembers.

Reply:
@thgreasi that doesn't change anything does it? If v6 needs it then even if we waited for v7 then dropping it here would also drop it from v6. I'm assuming what should actually happen is this should already be moved to a v6 translation and that way it'll naturally disappear when someone queries v7

Reply:
@Page- You mean that we should drop it from here right away and just move it to the v6 translation?
I was only thinking about deprecating it here so that we pile up all breaking changes (when possible) till we release v7, but you have a point that this should be in v6.ts translations right away...

Reply:
Let me drop the deprecation notices and actually drop these in a follow-up.